pkg/agent/openflow/network_policy.go

```diff
@@ -2220,3 +2228,69 @@ func (f *featureNetworkPolicy) l7NPTrafficControlFlows() []binding.Flow {
 			Done(),
 	}
 }
+
+func (f *featureNetworkPolicy) loggingNPPacketFlowWithOperations(cookieID uint64, operations uint8) binding.Flow {
+	loggingOperations := binding.NewRegMark(PacketInOperationField, uint32(operations))
+	fb := L2ForwardingOutTable.ofTable.BuildFlow(priorityNormal).
+		MatchRegMark(SendToControllerRegMark, loggingOperations)
+	if f.ovsMetersAreSupported {
+		fb = fb.Action().Meter(PacketInMeterIDNP)
+	}
+	return fb.Action().SendToController([]byte{uint8(PacketInCategoryNP), operations}, false).
+		Cookie(cookieID).
+		Done()
+}
+
+func (f *featureNetworkPolicy) initLoggingFlows() []binding.Flow {
+	maxOperationValue := PacketInNPLoggingOperation + PacketInNPStoreDenyOperation + PacketInNPRejectOperation
+	flows := make([]binding.Flow, 0, maxOperationValue)
+	if !f.enableAntreaPolicy {
+		return flows
+	}
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	for operation := 1; operation <= maxOperationValue; operation++ {
+		flows = append(flows, f.loggingNPPacketFlowWithOperations(cookieID, uint8(operation)))
+	}
+	return flows
+}
+
+func (f *featureNetworkPolicy) loggingGroups() []binding.OFEntry {
+	var groups []binding.OFEntry
+	if !f.enableAntreaPolicy {
+		return groups
+	}
+	candidateTables := []*Table{EgressRuleTable, EgressMetricTable, IngressRuleTable, IngressMetricTable}
+	if f.enableMulticast {
+		candidateTables = append(candidateTables, MulticastEgressMetricTable, MulticastIngressMetricTable)
+	}
+	for _, nextTable := range candidateTables {
+		groupKey := fmt.Sprintf("%s%d", loggingAndResubmitTableGroupPrefix, nextTable.GetID())
```

Comment:
Is the prefix really required? This is an internal cache, and all keys will have same prefix, right?

Reply:
removed.